src/utils/mongo-db/Models/user_model.ts

```diff
@@ -0,0 +1,210 @@
+import bcrypt from 'bcrypt';
+import mongoose, { Document, Schema } from 'mongoose';
+
+
+interface IUser extends Document {
+    firstName: string;
+    lastName: string;
+    username: string;
+    email: string;
+    password: string;
+}
+
+const userSchema: Schema = new Schema<IUser>({
+    firstName: {
+        type: String,
+    },
+    lastName: {
+        type: String,
+    },
+    username: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    password: {
+        type: String,
+        required: true,
+    },
+});
+
+
+/**
+ * Hashes the user's password before saving it into the database.
+ *
+ * @this {IUser} - the user document
+ * @param {Function} next - the callback function to call after hashing the password
+ */
+userSchema.pre<Document>('save', function(this: IUser, next: Function) {
+    if(this.isModified('password')) {
+        bcrypt.hash(this.password, 8, (err: Error, hash: string) => {
```

Comment:
Eu usaria o hashSync em vez do hash por maior legibilidade.
Qualquer coisa como:

try {
  this.password = bcrypt.hashSync(this.password, Number(process.env.SALT_ROUNDS));
} catch(error) {
console.error('Error while hashing password:', error)
}

O que achas?